contracts/src/lib/LibSubnetActor.sol

```diff
@@ -98,16 +101,16 @@ library LibSubnetActor {
 
             LibStaking.setMetadataWithConfirm(validators[i], publicKeys[i]);
             LibStaking.setFederatedPowerWithConfirm(validators[i], powers[i]);
-
-            s.genesisValidators.push(Validator({addr: validators[i], weight: powers[i], metadata: publicKeys[i]}));
+            s.genesis.addValidator(validators[i]);
 
             unchecked {
                 ++i;
             }
         }
 
         s.bootstrapped = true;
-        emit SubnetBootstrapped(s.genesisValidators);
+        s.genesis.bootstrap(s.validatorSet);
```

Comment:
Should `bootstrapped` be moved into this new `SubnetGenesis` struct?

Reply:
yeah, actually I think a lot of the parameters can be moved to `SubnetGenesis` struct so that we have better encapsulation, now it's scattered around. If this struct is ok, then I will create more PRs to migrate the fields one by one.